File: structural/composite-example.js
class Principal {
    static #name;

    constructor(name) {
        Principal.#name = name;
    };


    toString() {
        return Principal.#name;
    }
}

// ----------------------------------------------------------------------
class ClassRoom {
    static #name;

    constructor(name) {
        ClassRoom.#name = name;
    };


    toString() {
        return ClassRoom.#name;
    }
}

// ----------------------------------------------------------------------
class School {
    static #name;
    static #principal;
    static #classRooms;

    constructor(name, manager_name, class_count) {
        School.#name = name;
        School.#principal = new Principal(manager_name);
        School.#classRooms = [];
        for (let i in class_count) {
            School.#classRooms.push(new ClassRoom('class #' + str(i)));
        }
    }

    toString() {
        let school_as_str = `name: ${School.#name}, principal: ${School.#principal.toString()}\n`;
        for (let c of School.#classRooms) {
            school_as_str += ' ' + c.toString() + '\n';
        }
        return school_as_str;
    }
}


// ----------------------------------------------------------------------

let school = new School('Thorn Hill School', 'Mr. bean', 5);
console.log(school.toString());

File: structural/composite-example.py
class Principal:
    def __init__(self, name):
        self.__name = name

    def to_str(self):
        return self.__name


class ClassRoom:
    def __init__(self, name):
        self.__name = name

    def to_str(self):
        return self.__name


class School:
    def __init__(self, name, manager_name, class_count):
        self.__name = name
        self.__principal = Principal(manager_name)
        self.__class_rooms = []
        for i in range(class_count):
            self.__class_rooms.append(ClassRoom('class #' + str(i)))

    def to_str(self):
        school_as_str = f'name: {self.__name}, principal:{self.__principal.to_str()}'
        school_as_str += '\n'
        for c in self.__class_rooms:
            school_as_str += ' ' + c.to_str() + '\n'
        return school_as_str


if __name__ == "__main__":

    school = School('Thorn Hill School', 'Mr. bean', 5)
    print(school.to_str())
